W-12487830-search-patterns-cv (#130)
mulesoft/docs-rpa: rpa-builder/6.1/modules/ROOT/pages/adding-a-pattern-from-screen-capture.adoc

= Using Search Patterns in a Workflow 

> The *Capture* functionality provided in the Search Pattern, Pattern Select, and Pattern Gallery Action Steps enables you to create search patterns from a screenshot. Then, you can use these patterns in different elements of your Workflow. The Search Pattern, Pattern Select, and Pattern Gallery Action Steps also share the Pattern Strip module in the wizard, which provides a list of the existing patterns and their associated operations.

Reviewer: Should "Step" be lower-case instead of initial capped?

[[add-pattern-from-screen-capture]]
== Adding a Pattern from Screen Capture

You can create search patterns in the _Search Pattern_, _Pattern Select_, and _Pattern Gallery_ Action Steps by capturing a screenshot. The menu bars for these Action Steps include a *Capture* button (image:capture-button-icon.png[2%, 2%, The Capture button.]) for this purpose. 

Clicking *Capture* freezes the screen content like a screenshot and a small window with the title Capture Patterns appears. Then, you can select the search pattern by clicking and dragging to specify the area that contains the pattern. The selection surrounded with a blue rectangle is no longer transparent gray and you can move or resize it using the mouse or the keyboard. Press `Shift+Arrow key` to move the selection. Press `Ctrl+Arrow key` to resize the selection by moving the right and bottom edges.

image:search-pattern-capture.png[30%, 30%, The capture area of a search pattern.]

> You can select more than one search pattern by clicking and dragging multiple times. If the selected image detail is too small, not adequately differentiated, or it contains too few of the features needed to recognize a search pattern, the search pattern definition window shows red and a red cross appears.

Reviewer: Can you clarify what "the search pattern definition window shows red" means? Is the window red? Does the window have a red outline?

> image::incorrect-search-pattern.png[20%, 20%, The search pattern area showing a red cross indicating that the selection is not suitable as a search pattern.]

Reviewer: ```suggestion
image::incorrect-search-pattern.png[20%, 20%, The search pattern area shows a red cross, which indicates that the selection is not suitable as a search pattern.]
```

Clicking *Accept selection & continue* adds the applicable pattern to the wizard for the relevant Action Step. The wizard enables you to specify names and color spots for each of the captured patterns.

> By clicking the close button (*X*) in the upper right corner of the Capture Patterns window you can cancel the recording of a search pattern at any time.

Reviewer: Click the... to cancel...

== The Capture Patterns Window 

image::capture-patterns-window.png[30%, 30%, The Capture Patterns window]

* *Unfreeze screen* 
+
Releases the frozen screen so that changes can be made on the screen again.
* *Define Search Area* 
+

> Enables you to reduce the Search Area from full screen to a smaller area by clicking and dragging the mouse. This area is then given a red border and goes transparent red. When defining the Search Area you also have the option of unfreezing and refreezing the screen.

Reviewer: ```suggestion
Enables you to reduce the Search Area from full screen to a smaller area by clicking and dragging the mouse. This area is then given a red border and goes transparent red. When defining the Search Area, you also have the option of unfreezing and refreezing the screen.
```

* *Reuse last defined Search Area* 
+

> Enables you to reuse a search area defined in a previous capture. This option shows only if you already defined a search area when setting up the Workflow. 

Reviewer: Search area is in initial caps elsewhere in the text.

* *Accept selection & continue* 
+
Adds the selection made into the wizard for the corresponding Action Step and then returns to that Action Step.

After you capture at least one search pattern, the *Delete selected pattern* option shows. This option removes the border around the previously selected pattern (or the last search area defined).

=== Unfreeze Screen 

When you click *Unfreeze screen* the Capture Patterns window changes as follows:

image::take-a-screenshot-window.png[30%, 30%, The Take a screenshot window that shows after clicking Unfreeze Screen.]

* *Capture screen* 
+
Freezes the screen image again so that the pattern can be captured. 
* *Capture screen with delay* 
+

> Starts the timer and after a delay the screen image refreezes. The Capture Patterns window reappears after the screen content freezes.

Reviewer: ```suggestion
Starts the timer and, after a delay, the screen image refreezes. The Capture Patterns window reappears after the screen content freezes.
```

By default, RPA Builder only captures the pattern features for search patterns and color plays no role in pattern-based searches. However, sometimes you may find that it is necessary to determine when the color in a certain position in the search pattern is correct. For example, the text of an error message needs to be in red, or a traffic light symbol needs to appear green rather than yellow or red. In order to guarantee this, there is an option to define a Color Spot for a Search Pattern. In the case of a Search Pattern with Color Spot, the pattern-based image search is successful only if the color in a defined position matches a predefined color.

When you hover a search pattern in the wizard for the relevant Action Step, an overlay menu bar shows at the end of the image. 

image:search-pattern-icons.png[20%, 20%, A search pattern with the overlay menu that shows the available actions.]

Click the *Color Spot Picker* (image:color-picker-icon.png[2%, 2%, The Color Spot Picker button.]) to open a window where you can select the color spot to use with the search pattern.

image:color-spot-picker-window.png[60%, 60%, The Color Spot Picker window.]

> As you move the mouse over the image, the *Live values* section updates showing the following information: 

Reviewer: As you move the mouse over the image, the *Live values* section shows the following information:

* *Color*
+
The color value, expressed in RGB color values (`R:255`, `G:255`, `B:255`), for the current image coordinates. 
* *Coordinates*
+

> The image coordinates based on the current position of the mouse, indicated in `x;y` coordinates being `0;0` the upper-left corner of the image.

Reviewer: Is there a way you can avoid positional words like "upper-left" for accessibility reasons? (I'm not sure if you can do that here.)

To define the current selection as a Color Spot, either click the left mouse button or press `Insert` or `Space` on the keyboard. 

To delete the Color Spot, use the `Delete` or `Backspace` key.

When you select a Color Spot, it shows in the *Selected Color Spot* section on the left side of the window. 

image:color-spot-picker-window-selected-color.png[60%, 60%, The Color Spot Picker window showing a selected color.]

The following settings lessen the restrictions on Color Spots:

* *Compare Radius (in pixels)* 
+
Determines the radius around the specified position for the Color Spot in which the specified color must be found. There is a graphical representation of this radius next to the input field.
* *Color Fuzziness (in bits)* 
+
Determines the color deviation which is permitted for each RGB color channel. Use this option to specify how many low-order digits in the binary 8-digit color code can be ignored when comparing the color.

[[pattern-strip]]
== The Pattern Strip Module

> Pattern Strip is a UI module that is used across all the wizards for the Search Pattern, Pattern Select, and Pattern Gallery Action Steps. The Pattern Strip consists of a list of Search Patterns, the associated options, and a menu bar. The menu bar buttons vary according to the Action Step and the quantity and selection status of the search patterns.

Reviewer: "across all wizards" or across "all of the wizards"

.Pattern Strip for the Pattern Gallery Action Step:
image:pattern-strip-pattern-gallery.png[75%, 75%, The Pattern Strip module for the Pattern Gallery Action Step.]

.Pattern Strip for the Pattern Select Action Step:
image:pattern-strip-pattern-select.png[75%, 75%, The Pattern Strip module for the Pattern Select Action Step.]

Use the mouse to click and select search patterns. A white check mark appears on a blue background in the top-right corner to highlight a selected item. Click again to undo the selection.

Buttons for functions relating only to the selected patterns appear on the menu bar only if at least one pattern is selected.

=== Menu Bar

> The Pattern Strip menu bar has general buttons that show in all related Action Steps and buttons that are specific to the Search Pattern, Pattern Select, and Pattern Gallery Action Steps. The following table shows the button images, names, descriptions, and also in which of the related Action Steps they are available:  

Reviewer: The following table shows the button images, names, and descriptions, and the Action Steps in which they are available:
